inc/options/class-wpseo-options.php

```diff
@@ -171,25 +171,41 @@ public static function get_option_names() {
 	 * @return  array  Array combining the values of (nearly) all the options
 	 */
 	public static function get_all() {
-		$all_options  = array();
-		$option_names = self::get_option_names();
+		return self::get( self::get_option_names() );
+	}
+
+	/**
+	 * Retrieve one or more options for the SEO plugin.
+	 *
+	 * @param array|string $option_names The option name or an array of option names of the options you want to get.
+	 *
+	 * @static
+	 * @return  array  Array combining the values of the requested options
+	 */
+	public static function get( $option_names ) {
+		$options = array();
+
+		if ( is_string( $option_names ) && $option_names !== '' ) {
+			$option_names = array( $option_names );
```

Comment:
Be sure to filter validate the `$option_names` also because a value in array can contain a non-string value also.

`$option_names = array_filter( $option_names, 'is_string' );`